src/Modules/LearningTasks/Tutor.LearningTasks.Core/Domain/CaseStudy.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Tutor.BuildingBlocks.Core.Domain;
+
+namespace Tutor.LearningTasks.Core.Domain;
+
+public class CaseStudy : Entity
+{
+    public string Name { get; private set; }
+    public string Description { get; private set; }
+
+    public CaseStudy(string name, string description)
```

Comment:
Will the case study and domain model be loaded on demand or with every loading of the learning task? If it is on demand, I would treat these as separate single-entity aggregates, remove the association in the Learning Task and add in these two classes `LearningTaskId`.

Reply:
The case study and domain model will be loaded on demand, so I will treat these as separate single-entity aggregates. I will remove the association in the Learning Task and add LearningTaskId to these two classes.